lib/ramble/docs/tutorials/using_zips_and_matrices.rst

```diff
@@ -0,0 +1,354 @@
+
+Create a Workspace
+------------------
+
+To begin with, you need a workspace to configure the experiments. This can be
+created with the following command:
+
+.. code-block:: console
+
+    $ ramble workspace create zips_and_matrices_wrf
+
+
+Activate the Workspace
+----------------------
+
+Several of Ramble's commands require an activated workspace to function
+properly. Activate the newly created workspace using the following command:
+(NOTE: you only need to run this if you do not currently have the workspace
+active).
+
+.. code-block:: console
+
+    $ ramble workspace activate zips_and_matrices_wrf
+
+Configure Experiment Definitions
+--------------------------------
+
+To being with, you need to configure the workspace. The workspace's root
+location can be seen under the ``Location`` output of:
+
+.. code-block:: console
+
+    $ ramble workspace info
+
+Additionally, the files can be edited directly with:
+
+.. code-block:: console
+
+    $ ramble workspace edit
+
+Within the ``ramble.yaml`` file, write the following contents, which are the
+final configuration from the previous tutorial.
+
+.. code-block:: YAML
+
+    ramble:
+      variables:
+        processes_per_node: 4
+        n_ranks: '{processes_per_node}*{n_nodes}'
+        batch_submit: '{execute_experiment}'
+        mpi_command: mpirun -n {n_ranks}
+      applications:
+        wrfv4:
+          workloads:
+            CONUS_12km:
+              experiments:
+                scaling_{n_nodes}:
+                  variables:
+                    n_nodes: [1, 2, 4]
+      spack:
+        concretized: true
+        packages:
+          gcc9:
+            spack_spec: gcc@9.3.0
+          intel-mpi:
+            spack_spec: intel-mpi@2018.4.274
+            compiler: gcc9
+          wrfv4:
+            spack_spec: wrf@4.2 build_type=dm+sm compile_type=em_real nesting=basic ~chem
+              ~pnetcdf
+            compiler: gcc9
+        environments:
+          wrfv4:
+            packages:
+            - intel-mpi
+            - wrfv4
+
+The above configuration will execute 3 experiments, comprising a basic scaling
+study on three different sets of nodes.
+
+We will now expand this to perform more experiments using the zip
+(:ref:`ramble-explicit-zips`) and matrix (:ref:`ramble-matrix-logic`)
+functionality in Ramble.
+
+Construct Platforms Zip
+-----------------------
+
+For the purposes of this tutorial, you will construct a zip representing
+multiple platforms. The platforms will differ by their value of the
+``processes_per_node`` variable.
+
+Zips are explicit groupings of variables, that are combined into a larger
+variable set. A zip is defined by a list of variable definitions, where each
+individual variable is a list / vector variable and all variables are the same
+length. As an example, imagine we had the following variable / zip definitions:
+
+.. code-block:: YAML
+
+    variables:
+      platform: ['platform1', 'platform2']
```

Comment:
Is there something other than platform we can have as our second dimension? I like that it is a nice concrete and relatable use case, but I don't like how easy it is for a user to go adapt this and get very different behavior depending on the target spack auto detects

Reply:
It should actually do something right now. The platform is not overly interesting in this case but the platforms have different PPN (so n_ranks should change on the various platforms).

However, I'm open to changing the zip to be something else, I just don't have any great ideas about what it should be.